app/handlers/sessions_callback.rb

```diff
@@ -102,24 +102,24 @@ def handle
         run(TransferAuthentications, authentication, current_user)
         status = (current_user.is_temp ? :new_user : :returning_user)
       else
-        outcome = run(CreateUserFromOmniauth, @auth_data)
+        outcome = run(CreateUserFromOmniauthData, @data)
         new_user = outcome.outputs[:user]
         run(TransferAuthentications, authentication, new_user)
-        sign_in(new_user)
+        sign_in!(new_user)
         status = :new_user
       end
 
     end
 
     if this_authentication_is_new
-      run(TransferOmniauthInformation, @auth_data, current_user)
+      run(AddEmailFromOmniauthData, @data, current_user)
```

Comment:
I'm ok with the name -- in the future it could transfer mobile phones too.  It just moves selected information from the resulting omniauth to the user.  Just so happens that the only selected information so far is email.